File: tests/fixtures/Sample.java
/**
 * Sample Java file for testing parser functionality.
 * This demonstrates various Java language features.
 */
package com.example.testing;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;
import java.time.LocalDateTime;
import java.io.*;

/**
 * Sample interface for testing
 */
public interface Processor<T> {
    /**
     * Process an item
     * @param item The item to process
     * @return Processed result
     */
    T process(T item);
}

/**
 * Sample abstract class demonstrating inheritance
 */
abstract class AbstractService {
    protected String serviceName;

    public AbstractService(String name) {
        this.serviceName = name;
    }

    /**
     * Abstract method to be implemented by subclasses
     * @return Service status
     */
    public abstract boolean isRunning();
}

/**
 * Main sample class demonstrating various Java features
 *
 * @author Test Author
 * @version 1.0
 * @since 2024
 */
@SuppressWarnings("unused")
public class Sample extends AbstractService implements Processor<String>, Serializable {
    private static final long serialVersionUID = 1L;
    private static int instanceCount = 0;

    private final Long id;
    private String name;
    private LocalDateTime createdAt;
    private List<String> tags;

    /**
     * Default constructor
     */
    public Sample() {
        this(null, "default");
    }

    /**
     * Constructor with parameters
     *
     * @param id The unique identifier
     * @param name The name of the sample
     */
    public Sample(Long id, String name) {
        super("SampleService");
        this.id = id;
        this.name = name;
        this.createdAt = LocalDateTime.now();
        this.tags = new ArrayList<>();
        instanceCount++;
    }

    /**
     * Get the ID
     * @return The ID value
     */
    public Long getId() {
        return id;
    }

    /**
     * Get the name
     * @return The name value
     */
    public String getName() {
        return name;
    }

    /**
     * Set the name
     * @param name The new name
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * Process a string according to interface contract
     *
     * @param item The string to process
     * @return Processed string
     * @throws IllegalArgumentException if item is null
     */
    @Override
    public String process(String item) {
        if (item == null) {
            throw new IllegalArgumentException("Item cannot be null");
        }
        return item.toUpperCase();
    }

    /**
     * Check if service is running
     * @return Always returns true for this sample
     */
    @Override
    public boolean isRunning() {
        return true;
    }

    /**
     * Static method to get instance count
     * @return Number of instances created
     */
    public static int getInstanceCount() {
        return instanceCount;
    }

    /**
     * Generic method example
     * @param <T> Type parameter
     * @param list List to process
     * @return First element or null
     */
    public <T> T getFirst(List<T> list) {
        return list.isEmpty() ? null : list.get(0);
    }

    /**
     * Method with varargs
     * @param values Variable number of strings
     * @return Concatenated string
     */
    public String concatenate(String... values) {
        return String.join(", ", values);
    }

    /**
     * Method demonstrating Java 8 features
     * @param items List of items to filter
     * @param prefix Prefix to match
     * @return Filtered list
     */
    public List<String> filterByPrefix(List<String> items, String prefix) {
        return items.stream()
            .filter(item -> item.startsWith(prefix))
            .map(String::toLowerCase)
            .collect(Collectors.toList());
    }

    /**
     * Method returning Optional
     * @param index Index to retrieve
     * @return Optional containing the tag or empty
     */
    public Optional<String> getTag(int index) {
        if (index >= 0 && index < tags.size()) {
            return Optional.of(tags.get(index));
        }
        return Optional.empty();
    }

    /**
     * toString implementation
     * @return String representation
     */
    @Override
    public String toString() {
        return String.format("Sample[id=%d, name=%s]", id, name);
    }

    /**
     * Inner class example
     */
    public static class Builder {
        private Long id;
        private String name;

        public Builder withId(Long id) {
            this.id = id;
            return this;
        }

        public Builder withName(String name) {
            this.name = name;
            return this;
        }

        public Sample build() {
            return new Sample(id, name);
        }
    }

    /**
     * Enum example
     */
    public enum Status {
        ACTIVE("Active"),
        INACTIVE("Inactive"),
        PENDING("Pending");

        private final String displayName;

        Status(String displayName) {
            this.displayName = displayName;
        }

        public String getDisplayName() {
            return displayName;
        }
    }
}

/**
 * Another class in the same file
 */
@Deprecated
class Helper {
    /**
     * Utility method
     * @param value Value to check
     * @return true if valid
     */
    public static boolean isValid(String value) {
        return value != null && !value.trim().isEmpty();
    }
}

File: tests/fixtures/sample.php
<?php
/**
 * Sample PHP file for testing parser functionality.
 * This demonstrates various PHP language features.
 */

namespace Sample\Testing;

use DateTime;
use Exception;
use Sample\Base\AbstractClass;
use Sample\Utils\Helper as UtilHelper;

/**
 * Sample interface for testing
 */
interface SampleInterface {
    public function process(string $data): bool;
}

/**
 * Sample trait for testing
 */
trait LoggerTrait {
    /**
     * Log a message
     *
     * @param string $message The message to log
     * @return void
     */
    public function log(string $message): void {
        echo "[LOG] " . $message . PHP_EOL;
    }
}

/**
 * A sample abstract class for testing inheritance
 */
abstract class BaseClass {
    protected string $name;

    abstract public function getName(): string;
}

/**
 * Main sample class demonstrating various PHP features
 *
 * @package Sample\Testing
 * @author Test Author
 */
class SampleClass extends BaseClass implements SampleInterface {
    use LoggerTrait;

    private DateTime $createdAt;
    private static int $instanceCount = 0;

    /**
     * Constructor for SampleClass
     *
     * @param string $name The name to set
     * @param DateTime|null $date Optional date parameter
     */
    public function __construct(string $name, ?DateTime $date = null) {
        $this->name = $name;
        $this->createdAt = $date ?? new DateTime();
        self::$instanceCount++;
    }

    /**
     * Get the name property
     *
     * @return string
     */
    public function getName(): string {
        return $this->name;
    }

    /**
     * Process data according to interface contract
     *
     * @param string $data Data to process
     * @return bool Success status
     * @throws Exception When processing fails
     */
    public function process(string $data): bool {
        $this->log("Processing: " . $data);

        if (empty($data)) {
            throw new Exception("Data cannot be empty");
        }

        return true;
    }

    /**
     * Static method to get instance count
     *
     * @return int Number of instances created
     */
    public static function getInstanceCount(): int {
        return self::$instanceCount;
    }

    /**
     * Magic method for string representation
     *
     * @return string
     */
    public function __toString(): string {
        return sprintf("SampleClass[name=%s]", $this->name);
    }

    /**
     * Generator function example
     *
     * @param int $max Maximum number to generate
     * @return iterable
     */
    public function generateNumbers(int $max): iterable {
        for ($i = 0; $i <= $max; $i++) {
            yield $i;
        }
    }
}

/**
 * Standalone function for testing
 *
 * @param array $items Array of items to process
 * @param callable $callback Callback to apply
 * @return array Processed items
 */
function processItems(array $items, callable $callback): array {
    return array_map($callback, $items);
}

/**
 * Another standalone function with default parameter
 *
 * @param string $prefix Prefix to add
 * @param string $suffix Suffix to add
 * @return string Combined string
 */
function formatString(string $prefix = "START", string $suffix = "END"): string {
    return $prefix . " - content - " . $suffix;
}

// Example of closure
$multiplier = function(int $x, int $y): int {
    return $x * $y;
};

// Arrow function (PHP 7.4+)
$squared = fn($n) => $n * $n;
